platform/run_image.go

```diff
@@ -0,0 +1,78 @@
+package platform
+
+import (
+	"encoding/json"
+	"errors"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+)
+
+const (
+	TargetLabel                = "io.buildpacks.target"
+	OSDistributionNameLabel    = "io.buildpacks.distribution.name"
+	OSDistributionVersionLabel = "io.buildpacks.distribution.version"
+)
+
+type TargetMetadata struct {
+	ID           string          `toml:"id"`
+	OS           string          `toml:"os"`
+	Arch         string          `toml:"arch"`
+	ArchVariant  string          `toml:"variant"`
+	Distribution *OSDistribution `toml:"distribution"`
+}
+
+type OSDistribution struct {
```

Comment:
here too --  what's your guidance on de/coupling with the bp_descriptor.go file?    My gut still says that they're reading & writing the same things, so it's better to have them using the same structs.  But I also think it's somewhere between possible and likely that I'm missing something..

Reply:
In this case the platform spec and the buildpack spec happen to match each other (and happily so), but there isn't any technical reason that they are tied together. By contrast where for example `platform.BuildMetadata` depends on `buildpack.Label` that schema is directly coming from the buildpack spec and how buildpacks define labels. At least, that is my rationalization.